### PR TITLE
Script to auto start worker based on start date

### PR DESCRIPTION
**Pull request by Ram81**
### Description

This PR adds a script to auto launch/stop submission workers for challenges with start and end date based on current date. This will keep evalai admin out of the loop for starting/stopping challenge worker of every challenge. The script will run as a cron job everyday with decided schedule.

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@Ram81 Please fix the travis build and let me know when this is ready for review.

File: scripts/workers/auto_start_workers.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    for challenge in response["results"]:
+        challenge_id = challenge["id"]
+        workers = challenge["workers"]
+        is_docker_based = challenge["is_docker_based"]
+        challenge_start_date = parser.parse(challenge["start_date"])
+        challenge_end_date = parser.parse(challenge["end_date"])
+        current_date = utc.localize(datetime.now())
+
+        print('Start submission worker for challenge id {}'.format(challenge_id))
+        if not is_docker_based:
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:46]: can we add a check for `remote_challenges` as well.

File: scripts/workers/auto_start_workers.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import pytz
+import requests
+
+from datetime import datetime
+from dateutil import parser
+
+utc = pytz.UTC
+evalai_endpoint = os.environ.get("API_HOST_URL")
+authorization_header = {
+    "Authorization": "Token {}".format(os.environ.get("AUTH_TOKEN"))
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:11]: Can we please change this?

File: scripts/workers/auto_start_workers.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+
+def start_worker(challenge_id):
+    start_worker_endpoint = "{}/api/challenges/{}/manage_worker/start".format(evalai_endpoint, challenge_id)
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:16]: Can we please create a `dict` of all the URLs and then fetch from there.

File: scripts/workers/auto_start_workers.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+def stop_worker(challenge_id):
+    stop_worker_endpoint = "{}/api/challenges/{}/manage_worker/stop".format(evalai_endpoint, challenge_id)
+    response = requests.post(stop_worker_endpoint, headers=authorization_header)
+    return response
+
+
+def get_all_challenges():
+    all_challenge_endpoint = "{}/api/challenges/challenge/present".format(evalai_endpoint)
+    print(all_challenge_endpoint)
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:29]: Please remove the print statement.

File: scripts/workers/auto_start_workers.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    response = requests.get(all_challenge_endpoint, headers=authorization_header)
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:30]: Can we please create a common function make get and post requests.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
@Ram81 How are we running this script?

File: scripts/workers/auto_start_workers.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import os
```
Comment (RishabhJain2018) [inline scripts/workers/auto_start_workers.py:1]: Can we please change the name of file to more meaningful as the script does both starting and stopping of workers.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by RishabhJain2018** [comment]
@Ram81 Can you please resolve the conflicts here as well.

**Review by RishabhJain2018 — APPROVED** [review]

File: scripts/monitoring/run_auto_start_or_stop_workers.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+path=$PWD
+auth_token=''
+api_host_url='https://staging.eval.ai'
```
Comment (RishabhJain2018) [inline scripts/monitoring/run_auto_start_or_stop_workers.sh:4]: We shouldn't be hardcoding this?

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
@Ram81 How will this script `run_auto_start_or_stop_workers.sh` be deployed? We should add it while starting the Django container?

File: scripts/monitoring/run_auto_start_or_stop_workers.sh (hunk 7 above)
Comment (Ram81) [reply scripts/monitoring/run_auto_start_or_stop_workers.sh:4]: This is just the default argument. When the script is deployed as crontab we will add separate endpoints as shell script params in crontab

**Comment by Ram81** [comment]
@RishabhJain2018 we will deploy it as a crontab.

File: scripts/monitoring/run_auto_start_or_stop_workers.sh (hunk 7 above)
Comment (RishabhJain2018) [reply scripts/monitoring/run_auto_start_or_stop_workers.sh:4]: let's keep it blank then with the comments to avoid any future confusion.

**Comment by RishabhJain2018** [comment]
okay, but we should ideally not decouple components. It's fine for now but we should move it to the Django container.

File: scripts/monitoring/run_auto_start_or_stop_workers.sh (hunk 7 above)
Comment (RishabhJain2018) [reply scripts/monitoring/run_auto_start_or_stop_workers.sh:4]: @Ram81 Can you please make this change.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by Ram81** [comment]
@RishabhJain2018 done

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**